torch/ao/quantization/fx/prepare.py

```diff
@@ -278,58 +281,105 @@ def get_target_activation_dtype_for_node(
         is_getitem = node.op == 'call_function' and \
             node.target == operator.getitem
         if is_getitem:
-            return torch.float
+            return {
+                "input_activation_dtype": torch.float,
+                "output_activation_dtype": torch.float,
+            }
 
         # get qconfig to determine the eventual dtype of this node
         if qconfig is not None:
             if qhandler is not None and qhandler.input_output_observed() and qhandler.is_output_quantized(qconfig):
                 act_dtype, weight_dtype, act_compute_dtype = \
                     get_qconfig_dtypes(qconfig)
-                return act_dtype
-            else:
-                return torch.float
-        else:
-            return torch.float
+                bias_dtype = torch.float16 \
+                    if act_dtype == torch.float16 and weight_dtype == torch.float16 \
+                    else torch.float
+                return {
+                    "input_activation_dtype": act_dtype,
+                    "weight_dtype": weight_dtype,
+                    "bias_dtype": bias_dtype,
+                    "output_activation_dtype": act_dtype,
+                }
+        return {
+            "input_activation_dtype": torch.float,
+            "output_activation_dtype": torch.float,
+        }
 
     elif node.op == 'get_attr':
-        return torch.float
+        return {
+            "input_activation_dtype": torch.float,
+            "output_activation_dtype": torch.float,
+        }
 
     elif node.op == 'output':
         if outputs_seen_counter in output_quantized_idxs:
-            return torch.quint8
+            return {
+                "input_activation_dtype": torch.quint8,
+                "output_activation_dtype": torch.quint8
+            }
         else:
             # if dtype is fp32 (default), do nothing
             # note: other dtypes are not supported
-            return torch.float
+            return {
+                "input_activation_dtype": torch.float,
+                "output_activation_dtype": torch.float,
+            }
 
     else:
         raise AssertionError(f'need to handle {node.format_node()}')
 
+def get_arg_target_dtype(
+    arg: Node,
+    modules: Dict[str, torch.nn.Module],
+    node_name_to_target_dtype: Dict[str, Dict[str, Optional[torch.dtype]]],
+) -> Optional[torch.dtype]:
+    """ Get the target output activation dtype for
+    the argumnet in the original graph, skipping inserted observers
+    We are assuming that the observers are inserted correctly, and the dtype for
+    argument in quantized graph will match what is specified by the qconfig
+    """
+    assert isinstance(arg, Node)
+    if is_activation_post_process_node(arg, modules):
+        observed_arg = arg.args[0]
+        assert isinstance(observed_arg, Node), "Currently we only support observing Node"
+        return node_name_to_target_dtype[observed_arg.name]["output_activation_dtype"]
```

Comment:
are there any potential cases where we need to skip more than one observer?  I can't think of any, just wondering.

Reply:
No, we don't need to skip multiple observers, there can only be at most one observer based on our implementation the reason is that for each of the "Value" in the graph, we only insert observer when we first create the Value, and then when we use the value (input for some node), for the first case we will have 0 observers, for the second one we have at most 1 observer